src/com/google/javascript/jscomp/js/es6/util/inherits.js

```diff
@@ -49,25 +49,31 @@
  *
  * @param {!Function} childCtor Child class.
  * @param {!Function} parentCtor Parent class.
+ * @param {boolean} es5Out language out is ES5
  */
-$jscomp.inherits = function(childCtor, parentCtor) {
-  /** @constructor */
-  function tempCtor() {}
-  tempCtor.prototype = parentCtor.prototype;
-  childCtor.superClass_ = parentCtor.prototype;
-  childCtor.prototype = new tempCtor();
-  /** @override */
-    childCtor.prototype.constructor = childCtor;
+$jscomp.inherits = function(childCtor, parentCtor, es5Out) {
+  if (es5Out === true && Object.setPrototypeOf) {
```

Comment:
Why do we need this complexity? It only adds more features when it works. It should not break any existing code.

This is a well known restriction to JS devs. The fallback is to copy the descriptors exactly like we do now (which gives most of the support). There are only some edge cases that fail.